src/crimson/osd/pg.cc

```diff
@@ -824,7 +824,12 @@ PG::do_osd_ops(
         peering_state.get_info().last_user_version);
       return do_osd_ops_iertr::make_ready_future<Ref<MOSDOpReply>>(std::move(reply));
     }
-  ).finally([ox_deleter=std::move(ox)] {});
+  ).safe_then_unpack_interruptible(
+    [ox_deleter=std::move(ox)](auto submitted, auto all_completed) mutable {
```

Comment:
@rzarzynski @xxhdx1985126 shall we use shared_ptr or keep using the deleter-and-then-end approach?

Reply:
i am fine either way.

Reply:
I now think that maybe the deleter-and-then-end apprach is better, as it involves less object constructions.